src/fdb5/database/DB.cc

```diff
@@ -103,13 +104,19 @@ eckit::DataHandle *DB::retrieve(const Key& key) {
     return nullptr;
 }
 
-void DB::archive(const Key& key, const void* data, eckit::Length length) {
+void DB::archive(const Key& key, const void* data, eckit::Length length, const Key& internalKey, ArchiveCallback callback) {
 
     CatalogueWriter* cat = dynamic_cast<CatalogueWriter*>(catalogue_.get());
     ASSERT(cat);
 
     const Index& idx = cat->currentIndex();
-    cat->archive(key, store().archive(idx.key(), data, length));
+    std::unique_ptr<FieldLocation> location(store().archive(idx.key(), data, length));
+
+    if (callback) {
```

Comment:
This makes me wonder if we do better to have a NOP function, rather than a control-flow if() in the pathway.